web/src/views/DandisetLandingView/ListingComponent.vue

```diff
@@ -80,24 +30,106 @@ export default {
       type: [Object, Number, String, Array],
       required: true,
     },
-    root: {
-      type: Boolean,
-      default: false,
+    field: {
+      // The key in the parent object
+      type: String,
+      required: true,
     },
   },
   computed: {
+    renderComponent() {
+      // Determines which component to render
+
+      // Temporary for Girder
+      if (!toggles.DJANGO_API) {
+        switch (this.field) {
+          case 'access':
+            return ObjectComponent;
+          case 'contributors':
+          case 'publications':
+          case 'associatedData':
+          case 'associated_anatomy':
+          case 'sponsors':
+            return ObjectArray;
+          default:
+            break;
+        }
+      }
+
+      switch (this.field) {
+        case 'wasGeneratedBy':
+          return ObjectComponent;
+        case 'about':
+        case 'access':
+        case 'relatedResource':
+        case 'variableMeasured':
+          return ObjectArray;
+        case 'dandisetStats':
+          return DandisetStats;
+        case 'contributor':
+          return Contributor;
+        case 'keywords':
+          return SimpleArray;
+        default:
+          break;
+      }
+
+      switch (this.schema.type) {
+        case 'array':
+          if (this.schema.type.items && this.schema.type.items.type === 'object') {
+            return ObjectArray;
+          }
+          return SimpleArray;
+
+        default:
+          break;
+      }
+
+      return Primitive;
+    },
+    options() {
+      // General purpose options for different components
+      const { primaryKey } = this;
+      return {
+        primaryKey,
+      };
+    },
     primaryKey() {
-      switch (this.schema.title) {
-        case 'Access':
+      // Doesn't apply to all components
+
+      // Temporary for Girder
+      if (!toggles.DJANGO_API) {
+        switch (this.field) {
+          case 'publications':
+            return 'url';
+          default:
+            break;
+        }
+      }
+
+      switch (this.field) {
+        case 'access':
           return 'status';
-        case 'Publications':
+        case 'relatedResource':
           return 'url';
-        case 'Organism':
-          return 'species';
         default:
           return 'name';
       }
     },
+    componentClass() {
+      let className = '';
+
+      switch (this.renderComponent) {
+        case ObjectComponent:
+          className += 'ml-4';
+          break;
+
+        default:
+          break;
+      }
+
+      return className;
+    },
```

Comment:
This is a little extra. Is this designed to be extensible, or just leftover code?

Reply:
Yes it's designed to be extensible. Similar to the other functions that contain these switches, this is designed to add styling classes to the specific rendering components if they need it.